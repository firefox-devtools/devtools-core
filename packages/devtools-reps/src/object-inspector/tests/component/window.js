/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

const { mount } = require("enzyme");
const React = require("react");
const { createFactory } = React;
const ObjectInspector = createFactory(require("../../index"));
const {
  createNode,
} = require("../../utils/node");

const gripWindowStubs = require("../../../reps/stubs/window");
const ObjectClient = require("../__mocks__/object-client");
const windowNode = createNode(
  null,
  "window",
  "windowpath",
  {value: gripWindowStubs.get("Window")}
);

function generateDefaults(overrides) {
  return {
    autoExpandDepth: 0,
    roots: [windowNode],
    createObjectClient: grip => ObjectClient(grip),
    ...overrides,
  };
}

describe("ObjectInspector - dimTopLevelWindow", () => {
  it("renders collapsed top-level window when dimTopLevelWindow is true", () => {
    // The window node should have the "lessen" class when collapsed.
    const props = generateDefaults({
      dimTopLevelWindow: true,
    });
    const oi = ObjectInspector(props);
    const wrapper = mount(oi);
    expect(wrapper.find(".node.lessen").exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  it("renders expanded top-level window when dimTopLevelWindow is true", () => {
    // The window node should not have the "lessen" class when expanded.
    const props = generateDefaults({
      dimTopLevelWindow: true,
      autoExpandDepth: 1,
    });
    const oi = ObjectInspector(props);
    const wrapper = mount(oi);
    expect(wrapper.find(".node.lessen").exists()).toBeFalsy();
    expect(wrapper).toMatchSnapshot();
  });

  it("renders collapsed top-level window when dimTopLevelWindow is false", () => {
    // The window node should not have the "lessen" class when dimTopLevelWindow is falsy.
    const props = generateDefaults();
    const oi = ObjectInspector(props);
    const wrapper = mount(oi);
    expect(wrapper.find(".node.lessen").exists()).toBeFalsy();
    expect(wrapper).toMatchSnapshot();
  });

  it("renders sub-level window", () => {
    // The window node should not have the "lessen" class when it is not at top level.
    const root = createNode(
      null,
      "root",
      "rootpath",
      [windowNode]
    );

    const props = generateDefaults({
      autoExpandDepth: 1,
      roots: [root],
      dimTopLevelWindow: true,
    });
    const oi = ObjectInspector(props);
    const wrapper = mount(oi);
    const nodes = wrapper.find(".node");
    const win = nodes.at(1);

    // Make sure we target the window object.
    expect(win.find(".objectBox-Window").exists()).toBeTruthy();
    expect(win.hasClass("lessen")).toBeFalsy();
    expect(wrapper).toMatchSnapshot();
  });
});
