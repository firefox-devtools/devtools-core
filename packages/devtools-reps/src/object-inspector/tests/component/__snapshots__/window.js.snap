// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectInspector - dimTopLevelWindow renders collapsed top-level window when dimTopLevelWindow is false 1`] = `
<ObjectInspector
  autoExpandDepth={0}
  getObjectProperties={[Function]}
  loadObjectProperties={[Function]}
  roots={
    Array [
      Object {
        "contents": Object {
          "value": Object {
            "actor": "server1.conn3.obj198",
            "class": "Window",
            "extensible": true,
            "frozen": false,
            "ownPropertyLength": 887,
            "preview": Object {
              "kind": "ObjectWithURL",
              "url": "about:newtab",
            },
            "sealed": false,
            "type": "object",
          },
        },
        "name": "window",
        "parent": null,
        "path": "windowpath",
        "type": Symbol(GRIP),
      },
    ]
  }
/>
`;

exports[`ObjectInspector - dimTopLevelWindow renders collapsed top-level window when dimTopLevelWindow is true 1`] = `
<ObjectInspector
  autoExpandDepth={0}
  dimTopLevelWindow={true}
  getObjectProperties={[Function]}
  loadObjectProperties={[Function]}
  roots={
    Array [
      Object {
        "contents": Object {
          "value": Object {
            "actor": "server1.conn3.obj198",
            "class": "Window",
            "extensible": true,
            "frozen": false,
            "ownPropertyLength": 887,
            "preview": Object {
              "kind": "ObjectWithURL",
              "url": "about:newtab",
            },
            "sealed": false,
            "type": "object",
          },
        },
        "name": "window",
        "parent": null,
        "path": "windowpath",
        "type": Symbol(GRIP),
      },
    ]
  }
/>
`;

exports[`ObjectInspector - dimTopLevelWindow renders expanded top-level window when dimTopLevelWindow is true 1`] = `
<ObjectInspector
  autoExpandDepth={1}
  dimTopLevelWindow={true}
  getObjectProperties={[Function]}
  loadObjectProperties={[Function]}
  roots={
    Array [
      Object {
        "contents": Object {
          "value": Object {
            "actor": "server1.conn3.obj198",
            "class": "Window",
            "extensible": true,
            "frozen": false,
            "ownPropertyLength": 887,
            "preview": Object {
              "kind": "ObjectWithURL",
              "url": "about:newtab",
            },
            "sealed": false,
            "type": "object",
          },
        },
        "name": "window",
        "parent": null,
        "path": "windowpath",
        "type": Symbol(GRIP),
      },
    ]
  }
/>
`;

exports[`ObjectInspector - dimTopLevelWindow renders sub-level window 1`] = `
<ObjectInspector
  autoExpandDepth={1}
  dimTopLevelWindow={true}
  getObjectProperties={[Function]}
  loadObjectProperties={[Function]}
  roots={
    Array [
      Object {
        "contents": Array [
          Object {
            "contents": Object {
              "value": Object {
                "actor": "server1.conn3.obj198",
                "class": "Window",
                "extensible": true,
                "frozen": false,
                "ownPropertyLength": 887,
                "preview": Object {
                  "kind": "ObjectWithURL",
                  "url": "about:newtab",
                },
                "sealed": false,
                "type": "object",
              },
            },
            "name": "window",
            "parent": null,
            "path": "windowpath",
            "type": Symbol(GRIP),
          },
        ],
        "name": "root",
        "parent": null,
        "path": "rootpath",
        "type": Symbol(GRIP),
      },
    ]
  }
/>
`;
